addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,130 @@
+import requests
+
+from odoo import models, fields, _, api
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_ro_edi_document_ids = fields.One2many(
+        comodel_name='l10n_ro_edi.document',
+        inverse_name='invoice_id',
+    )
+    l10n_ro_edi_active_document_id = fields.Many2one(
+        comodel_name='l10n_ro_edi.document',
+        compute='_compute_from_l10n_ro_edi_document_ids',
+    )
+    l10n_ro_edi_state = fields.Selection(
+        selection=[
+            ('invoice_sending', 'Sending'),
+            ('invoice_sending_failed', 'Error'),
+            ('invoice_sent', 'Sent'),
+        ],
+        string='E-Factura Status',
+        compute='_compute_from_l10n_ro_edi_document_ids_stored',
+        store=True,
+    )
+    l10n_ro_edi_message = fields.Char(compute='_compute_from_l10n_ro_edi_document_ids')
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_from_l10n_ro_edi_document_ids(self):
+        for move in self:
+            active_document = self.env['l10n_ro_edi.document'].search([('invoice_id', '=', move.id)], limit=1)
+            move.l10n_ro_edi_active_document_id = active_document
+            move.l10n_ro_edi_message = active_document.message
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_from_l10n_ro_edi_document_ids_stored(self):
+        for move in self:
+            active_document = self.env['l10n_ro_edi.document'].search([('invoice_id', '=', move.id)], limit=1)
+            move.l10n_ro_edi_state = active_document.state
+
+    @api.depends('l10n_ro_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        """ Prevent user to reset move to draft when there's an
+            active sending document or an OK response has been received """
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            if move.l10n_ro_edi_state in ('invoice_sending', 'invoice_sent'):
+                move.show_reset_to_draft_button = False
+
+    def _l10n_ro_edi_get_attachment_file_name(self):
+        self.ensure_one()
+        return f"{self.name.replace('/', '_')}.xml"
```

Comment:
You need specific data in the filename like `efactura_` because when downloading the attachments from the send & print, we want to avoid multiple attachments having the same name (trust me, it doesn't work).